pkg/build/api/v1/types.go

```diff
@@ -402,6 +402,11 @@ type SourceBuildStrategy struct {
 
 	// ForcePull describes if the builder should pull the images from registry prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// disableImplicitBuild prevents source-to-image from performing a Docker build operation
```

Comment:
@smarterclayton fair enough, but I want to be able to control this behavior on a per user basis so we need some way to convey to s2i what is/isn't allowed.  I'm open to suggestions, but I started by putting it on the build pod directly until I realized the build pods are managed by the controller which means they can't check permissions in a useful way.